controller/controller.go

```diff
@@ -509,14 +509,17 @@ func ProcessDelete(ctx context.Context, obj interface{}, resources []resource.In
 
 		err := r.EnsureDeleted(ctx, obj)
 		if err != nil {
+			ctx = unsetLoggerCtxValue(ctx, loggerKeyResource)
```

Comment:
What about just calling `defer unsetLoggerCtxValue(ctx, loggerKeyResource)` before the loop?

Reply:
Shouldn't we have to assign for the new context? such as 
```
	defer func() {
		ctx = unsetLoggerCtxValue(ctx, loggerKeyResource)
	}()
```